packages/examples/test/batteries/use-tools.tsx

```diff
@@ -0,0 +1,124 @@
+import * as AI from 'ai-jsx';
+import { UseTools } from 'ai-jsx/batteries/use-tools';
+import { ChatProvider } from 'ai-jsx/core/completion';
+import {
+  AssistantMessage,
+  FunctionCall,
+  ShowConversation,
+  UserMessage,
+  renderToConversation,
+} from 'ai-jsx/core/conversation';
+
+async function FakeChatCompletion({ children }: { children: AI.Node }, { render }: AI.ComponentContext) {
+  const nodes = await renderToConversation(children, render);
+  if (nodes.find((n) => n.type === 'functionCall')) {
+    return <AssistantMessage>DONE</AssistantMessage>;
+  }
+  return <FunctionCall name="myFunc" args={{ parameter: 'test parameter value' }} />;
+}
+
+describe('useTools', () => {
+  it('should give tools access to context', async () => {
+    const myContext = AI.createContext(0);
+
+    function MyTool({ parameter }: { parameter: string }, { getContext }: AI.ComponentContext) {
```

Comment:
I think it would be nice to have an example where the tool doesn't use JSX and is just a vanilla function.